src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -171,15 +177,26 @@ public boolean publish(String message, JSONArray attachments, String color) {
             } else {
                 url = "https://slack.com/api/chat.postMessage?token=" + populatedToken +
                         "&channel=" + roomId.replace("#", "") +
-                        "&link_names=1" +
-                        "&as_user=true";
+                        "&link_names=1";
                 if (threadTs.length() > 1) {
                     url += "&thread_ts=" + threadTs;
                 }
                 if (replyBroadcast) {
                     url += "&reply_broadcast=true";
                 }
                 try {
+                    if (StringUtils.isEmpty(iconEmoji) && StringUtils.isEmpty(username)) {
+                        url += "&as_user=true";
+                    }
+                    else {
+                        if (StringUtils.isNotEmpty(iconEmoji)) {
+                            url += "&icon_emoji=" + URLEncoder.encode(iconEmoji, StandardCharsets.UTF_8.name());
```

Comment:
May I suggest we start using handy URI?
You should use the Jenkins plugin version of it: https://github.com/jenkinsci/handy-uri-templates-2-api-plugin

However is the docs: https://github.com/damnhandy/Handy-URI-Templates
See usage here: https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/80/files

Reply:
See this comment: https://github.com/jenkinsci/slack-plugin/pull/577#discussion_r289592678